openpype/pipeline/colorspace.py

```diff
@@ -0,0 +1,464 @@
+        temporary_json_file = tempfile.NamedTemporaryFile(
+            mode="w", suffix=".json", delete=False
+        )
+        temporary_json_file.close()
+        temporary_json_filepath = temporary_json_file.name.replace(
+            "\\", "/"
+        )
+
+        yield temporary_json_filepath
+
+    except IOError as _error:
+        raise IOError(
+            "Not able to create temp json file: {}".format(
+                _error
+            )
+        )
+
+    finally:
+        # Remove the temporary json
+        os.remove(temporary_json_filepath)
+
+
+def get_ocio_config_script_path():
+    """Get path to ocio wrapper script
+
+    Returns:
+        str: path string
+    """
+    return os.path.normpath(
+        os.path.join(
+            PACKAGE_DIR,
+            "scripts",
+            "ocio_wrapper.py"
+        )
+    )
+
+
+def get_imageio_colorspace_from_filepath(
+    path, host_name, project_name,
+    config_data=None, file_rules=None,
+    project_settings=None,
+    validate=True
+):
+    """Get colorspace name from filepath
+
+    ImageIO Settings file rules are tested for matching rule.
+
+    Args:
+        path (str): path string, file rule pattern is tested on it
+        host_name (str): host name
+        project_name (str): project name
+        config_data (dict, optional): config path and template in dict.
+                                      Defaults to None.
+        file_rules (dict, optional): file rule data from settings.
+                                     Defaults to None.
+        project_settings (dict, optional): project settings. Defaults to None.
+        validate (bool, optional): should resulting colorspace be validated
+                                   with config file? Defaults to True.
+
+    Returns:
+        str: name of colorspace
+    """
+    if not any([config_data, file_rules]):
+        project_settings = project_settings or get_project_settings(
+            project_name
+        )
+        config_data = get_imageio_config(
+            project_name, host_name, project_settings)
+        file_rules = get_imageio_file_rules(
+            project_name, host_name, project_settings)
+
+    # match file rule from path
+    colorspace_name = None
+    for _frule_name, file_rule in file_rules.items():
+        pattern = file_rule["pattern"]
+        extension = file_rule["ext"]
+        ext_match = re.match(
+            r".*(?=.{})".format(extension), path
+        )
+        file_match = re.search(
+            pattern, path
+        )
+
+        if ext_match and file_match:
+            colorspace_name = file_rule["colorspace"]
+
+    if not colorspace_name:
+        log.info("No imageio file rule matched input path: '{}'".format(
+            path
+        ))
+        return None
+
+    # validate matching colorspace with config
+    if validate and config_data:
+        validate_imageio_colorspace_in_config(
+            config_data["path"], colorspace_name)
```

Comment:
The body of `validate_imageio_colorspace_in_config` isn't too complex and doesn't really warrant being an own function.

I'd rewrite this to:
```python
# pseudocode
if colorspace_name not in get_ocio_config_colorspaces(config_path):
    raise
```

Reply:
Yeah from the perspective of the actual state of code yes, but in the future, for example in settings we will want to validate user inputs.

Reply:
Think if the code is not validating the user inputs now, we should not anticipate it yet. This will just confuse later on.

Reply:
We are validating it already but only in one function.